src/core.js

```diff
@@ -23,32 +22,65 @@ async function loadPage (page, url, timeout, pageLoadSkipTimeout) {
   // set a higher number than the timeout option, in order to make
   // puppeteer’s timeout _never_ happen
   let waitingForPageLoad = true
-  const loadPagePromise = page.goto(url, { timeout: timeout + 1000 })
+  let pageLoadSkipTimeoutPromise = null
+  const loadPagePromise = new Promise((resolve, reject) => {
+    page
+      .goto(url, {
+        timeout: timeout + 1000
+      })
+      .then(response => {
+        // Reject exits the Promise.all call immediately and ensures that the pageload is always higher prio than timout
+        return resolve('loadPageResponse')
+      })
+      .catch(err => {
+        // Reject when error because we don't want an errored page
+        console.error(err)
+        return reject(err)
+      })
+  })
+
   if (pageLoadSkipTimeout) {
-    await Promise.race([
-      loadPagePromise,
-      new Promise(resolve => {
-        // instead we manually _abort_ page load after X time,
-        // in order to deal with spammy pages that keep sending non-critical requests
-        // (tracking etc), which would otherwise never load.
-        // With JS disabled it just shouldn't take that many seconds to load what's needed
-        // for critical viewport.
-        setTimeout(() => {
-          if (waitingForPageLoad) {
-            debuglog(
-              'page load waiting ABORTED after ' +
-                pageLoadSkipTimeout / 1000 +
-                's. '
-            )
-            resolve()
+    pageLoadSkipTimeoutPromise = new Promise((resolve, reject) => {
+      // enables us to control requests(stop, continue, modify)
+      page.on('response', async response => {
+        if (response.url === url) {
```

Comment:
what are you trying to do here? Why is it relevant that we got a response for the right url? As far as I can see it's not relevant for the pageLoadSkipTimeout, so what it is relevant for?

Reply:
this is also a leftover. I tried to find out why the error is occuring. So I just want to start the timeout when the page original url was requested and the response came in. In my opinion this is the real pageload. In this case we are talking about pageload and not response time. Response time is what your code measures.